File: .sdk-repos/orchestration-cluster-api-js/examples/secret.ts
// Compilable usage examples for secret operations.
// These examples are type-checked during build to guard against API regressions.

import { createCamundaClient } from '@camunda8/orchestration-cluster-api';

//#region ResolveSecrets
async function resolveSecretsExample() {
  const camunda = createCamundaClient();

  const result = await camunda.resolveSecrets({
    references: ['camunda.secrets.myApiToken', 'camunda.secrets.dbPassword'],
  });

  // Successfully resolved references are returned in `resolved`; references that
  // could not be resolved are returned in `errors`, each with a typed error code.
  // Never log a resolved value — it holds secret material. Pass it straight to the
  // consumer that needs it (HTTP client, DB driver, ...) instead.
  for (const resolved of result.resolved) {
    console.log(`Resolved ${resolved.reference} (value redacted)`);
    useSecret(resolved.value);
  }

  for (const error of result.errors) {
    console.log(`Failed to resolve ${error.reference}: ${error.code} - ${error.message}`);
  }
}

// Hands the resolved secret to whatever needs it, without logging it.
function useSecret(_value: string) {}
//#endregion ResolveSecrets

//#region ListSecrets
async function listSecretsExample() {
  const camunda = createCamundaClient();

  // The request body is reserved for future filtering options and currently
  // takes no properties.
  const result = await camunda.listSecrets({});

  // Only the references are returned — never the secret values. Use
  // `resolveSecrets` to fetch a value when one is actually needed.
  for (const reference of result.references) {
    console.log(`Secret available: ${reference}`);
  }
}
//#endregion ListSecrets

// Suppress "declared but never read"
void resolveSecretsExample;
void listSecretsExample;

File: .sdk-repos/orchestration-cluster-api-go/examples/secret.go
// Secret operations: resolve connector secret references to their values.
package examples

import (
	"context"
	"fmt"

	camunda "github.com/camunda/orchestration-cluster-api-go"
	openapi "github.com/camunda/orchestration-cluster-api-go/client"
)

func resolveSecretsExample(ctx context.Context, client *camunda.CamundaClient) error {
	// region ResolveSecrets
	// References take the form `camunda.secrets.<name>`.
	req := openapi.NewSecretResolveRequest([]string{"camunda.secrets.MY_API_KEY", "camunda.secrets.MY_TOKEN"})

	result, err := client.ResolveSecrets(ctx, *req)
	if err != nil {
		return err
	}
	for _, secret := range result.GetResolved() {
		fmt.Printf("%v = %v\n", secret.GetReference(), secret.GetValue())
	}
	// endregion ResolveSecrets
	return nil
}

func listSecretsExample(ctx context.Context, client *camunda.CamundaClient) error {
	// region ListSecrets
	// Returns only the references the caller is authorized to see — never values.
	result, err := client.ListSecrets(ctx)
	if err != nil {
		return err
	}
	for _, reference := range result.GetReferences() {
		fmt.Printf("%v\n", reference)
	}
	// endregion ListSecrets
	return nil
}
